pkg/xds/server/dataplane_status_tracker.go

```diff
@@ -120,6 +121,9 @@ func (c *dataplaneStatusTracker) OnStreamRequest(streamID int64, req *envoy.Disc
 	if state.dataplaneId == (core_model.ResourceKey{}) {
 		if id, err := core_xds.ParseProxyId(req.Node); err == nil {
 			state.dataplaneId = core_model.ResourceKey{Mesh: id.Mesh, Name: id.Name}
+			if err := util_proto.ToTyped(req.Node.Metadata, state.subscription.Version); err != nil {
+				xdsServerLog.Error(err, "failed to parse Node.Metadata out of DiscoveryRequest", "streamid", streamID, "req", req)
+			}
```

Comment:
what will happen if Envoy won't send the version in the metadata? Will this throw an error or will this just go through? It should be the second option.

also

```
failed to parse Node.Metadata out of DiscoveryRequest
```
->
```
xdsServerLog.Error(err, "failed to extract version out of the Envoy metadata", "streamid", streamID, "metadata", req.Node.Metadata)
```

Reply:
I tested the following setup:
1. kuma-cp (this branch)
2. kuma-dp (this branch) - sends version
3. kuma-dp (master) - doesn't send version
and there was no failure. Additionally I added unit tests that parse nil and random string and these cases also don't throw error. To be honest I don't know how can I simulate a scenario when it fails... Do you have any idea?

Reply:
nil / not nil scenario should be fine. Thanks for the extra test!